app/assets/javascripts/controllers/container_dashboard/util/dashboard-utils-factory.js

```diff
@@ -97,8 +97,28 @@ angular.module('miq.util').factory('dashboardUtilsFactory', function() {
   var parseDate = function(date) {
     myDate = Date.parse(date);
     return isNaN(myDate) ? date : myDate
-  }
+  };
+  var iconClassForProvider = function(provider) {
```

Comment:
> Uh, I think theyre different enough to keep apart but Im not keen on whats happening inside topology. What do you think @abonas ?

I think that there should be such mechanism to get the icon in png/text/unicode for models in ManageIQ, beyond containers or dashboard or topology scope. Because nearly every model in the project has an icon. Given that there is no such mechanism at the moment, and topology and dashboard use the different methods for retrieving icons (topology must use unicode), then I would not currently pursue this direction.